http/connection.go

```diff
@@ -91,11 +100,38 @@ func newHTTPConnection(endpoint string, config ConnectionConfig) (driver.Connect
 	if config.Transport != nil {
 		httpTransport, _ = config.Transport.(*http.Transport)
 	} else {
-		httpTransport = &http.Transport{}
+		httpTransport = &http.Transport{
+			// Copy default values from http.DefaultTransport
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+		}
 		config.Transport = httpTransport
 	}
-	if config.TLSConfig != nil && httpTransport != nil {
-		httpTransport.TLSClientConfig = config.TLSConfig
+	if httpTransport != nil {
+		if httpTransport.MaxIdleConnsPerHost == 0 {
+			// Raise the default number of idle connections per host since in a database application
+			// it is very likely that you want more than 2 concurrent connections to a host.
+			// We raise it to avoid the extra concurrent connections being closed directly
+			// after use, resulting in a lot of connection in `TIME_WAIT` state.
+			httpTransport.MaxIdleConnsPerHost = DefaultMaxIdleConnsPerHost
+		}
+		defaultMaxIdleConns := 3 * DefaultMaxIdleConnsPerHost
+		if httpTransport.MaxIdleConns > 0 && httpTransport.MaxIdleConns < defaultMaxIdleConns {
```

Comment:
It seems `MaxIdleConnsPerHost` is set to 0 here if nothing is set, is the same true for `MaxIdleConns`? If so, then we should write "== 0" here, otherwise we do not set out default if the user has not given anything.

Reply:
Behavior for `MaxIdleConns==0` is different. It means "unlimited"